osc-server/src/test/java/org/osc/core/broker/rest/client/k8s/KubernetesStatusApiTest.java

```diff
@@ -0,0 +1,167 @@
+        when(this.fabric8Client.componentstatuses()).thenThrow(new KubernetesClientException(""));
+
+        // Act.
+        this.api.isServiceReady();
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusIsNull_ServiceNotReady() throws Exception {
+        // Arrange.
+        mockComponentStatus(null);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionsIsNull_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus(null, null);
+        componentStatus.setConditions(null);
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusHealthyConditionNotFound_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("other_condition", "True");
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionFalse_ServiceNotReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("Healthy", "False");
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertFalse(isServiceReady);
+    }
+
+    @Test
+    public void testIsServiceReady_WhenComponentStatusConditionTrue_ServiceReady() throws Exception {
+        // Arrange.
+        ComponentStatus componentStatus = newComponentStatus("Healthy", "True");
+        mockComponentStatus(componentStatus);
+
+        // Act.
+        boolean isServiceReady = this.api.isServiceReady();
+
+        // Assert.
+        assertTrue(isServiceReady);
+    }
+
+    private void mockComponentStatus(ComponentStatus result) {
+        @SuppressWarnings("unchecked")
+        Resource<ComponentStatus, DoneableComponentStatus> componentStatusResource = Mockito.mock(Resource.class);
+        when(componentStatusResource.get()).thenReturn(result);
+        when(this.mixedOperationMock.withName("controller-manager")).thenReturn(componentStatusResource);
+    }
+
+    private ComponentStatus newComponentStatus(String conditionType, String conditionStatus) {
+        ComponentStatus componentStatus = new ComponentStatus();
+        List<ComponentCondition> conditions = new ArrayList<ComponentCondition>();
+
+        if (conditionType != null) {
+            ComponentCondition condition = new ComponentCondition();
```

Comment:
can be simplified into componentStatus.setConditions(Arrays.asList(new ComponentCondition(null, null, conditionType, conditionStatus)